pkg/distributor/distributor.go

```diff
@@ -347,14 +348,17 @@ func (d *Distributor) Push(ctx context.Context, req *logproto.PushRequest) (*log
 					bytes += len(e.Line)
 				}
 				validation.DiscardedBytes.WithLabelValues(validation.InvalidLabels, tenantID).Add(float64(bytes))
+				for _, tracker := range validationContext.customTrackerConfig.MatchTrackers(lbs) {
+					validation.DiscardedBytesCustom.WithLabelValues(validation.RateLimited, tenantID, tracker).Add(float64(bytes))
+				}
```

Comment:
we could loop once and save the trackers that the labelset matched, and then loop over those and update the discarded (and if needed rate limited) metrics

Reply:
That's a good point.

Reply:
No, you just have the same loop iteration later commented out 
```
/* We would have to filter all streams again here.
		for _, tracker := range 
```

If you're worried about iterating over all the trackers multiple times we could save the matching trackers once and then just iterate over those when we detect that we need to update the `DiscardedBytesCustom` metrics.

Reply:
Ah, I missed that comment thanks.

Reply:
Ok, this got a little messy. We cannot report on a labels basis if the labels cannot be parsed. We do want to report on labels basis when we hit the rate limit.